src/Analyser/TypeSpecifier.php

```diff
@@ -1107,39 +1107,6 @@ private function specifyTypesForConstantBinaryExpression(
 			));
 		}
 
-		if (
-			!$context->null()
-			&& $exprNode instanceof FuncCall
-			&& count($exprNode->getArgs()) === 1
-			&& $exprNode->name instanceof Name
-			&& in_array(strtolower((string) $exprNode->name), ['strlen', 'mb_strlen'], true)
-			&& $constantType instanceof ConstantIntegerType
```

Comment:
instead of comparison with `ConstantIntegerType` we now support `IntegerRangeType`